src/main/scala/utils/Clustering.scala

```diff
@@ -229,16 +299,20 @@ object Clustering {
     * @return
     *   either the total distance or [[Double.MaxValue]]
     */
-  private def totalDistance(list: List[Cluster]): Double = {
-    list.map { l => l.distances }.reduceOption { (a, b) => a + b } match {
-      case Some(value) => value
-      case None        => Double.MaxValue
+  private def totalDistance(list: Set[Cluster]): Double = {
```

Comment:
Please try if you find a naming that comply with sonar although it was named like this before.